db/migrate/20220504081145_add_composite_indexes_to_join_tables.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class AddCompositeIndexesToJoinTables < ActiveRecord::Migration[6.1]
+  def change
+    add_index :canonical_armors_enchantments,
+              %i[canonical_armor_id enchantment_id],
+              unique: true,
+              name:   'index_can_armors_enchantments_on_can_armor_id_and_ench_id'
```

Comment:
I had to make these awful shortened index names because Postgres limits them to 63 characters.